checks/evaluation/pinned_dependencies.go

```diff
@@ -90,66 +95,32 @@ func PinningDependencies(name string, c *checker.CheckRequest,
 				Snippet:     rr.Location.Snippet,
 				Remediation: generateRemediation(remediationMetadata, &rr),
 			})
-
-			// Update the pinning status.
-			updatePinningResults(&rr, &wp, pr)
 		}
+		// Update the pinning status.
+		updatePinningResults(&rr, &wp, pr)
 	}
 
 	// Generate scores and Info results.
-	// GitHub actions.
-	actionScore, err := createReturnForIsGitHubActionsWorkflowPinned(wp, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
-
-	// Docker files.
-	dockerFromScore, err := createReturnForIsDockerfilePinned(pr, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
-
-	// Docker downloads.
-	dockerDownloadScore, err := createReturnForIsDockerfileFreeOfInsecureDownloads(pr, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
-
-	// Script downloads.
-	scriptScore, err := createReturnForIsShellScriptFreeOfInsecureDownloads(pr, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
-
-	// Pip installs.
-	pipScore, err := createReturnForIsPipInstallPinned(pr, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
-
-	// Npm installs.
-	npmScore, err := createReturnForIsNpmInstallPinned(pr, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
+	var scores []*checker.ProportionalScoreWithWeight
+	// Go through all dependency types
+	// GitHub Actions need to be handled separately since they are not in pr
+	scores = append(scores, createScoreForGitHubActionsWorkflow(&wp)...)
```

Comment:
Is there a reason why they're not in `pr`, would it be easier to just add them to `pr`?
I do see how this might simplify the weighted 20/80 split though.

Reply:
Yes. I believe the reason is that `pr = map[checker.DependencyUseType]pinnedResult`. The `DependencyUseType` for GHA is `DependencyUseTypeGHAction`, but it does not distinct between `GitHub-owned` and `third-party` GHA. This comes like this from the `raw` part. So, we make the distinction at the `evaluation` using a separate structure `wp = worklowPinningResult`

To add GHAs into `pr`, I see the following options:
- branch `DependencyUseTypeGHAction` into `DependencyUseTypeGHAGitHubOwnedAction` and `DependencyUseTypeGHAThirdPartyAction`. This would move the distinction to `raw` instead of `evaluation`.
- modify `pr` to be `map[checker.DependencyUseType] (pinnedResult OR worklowPinningResult)`. Not sure if this is possible in Go.
- modify `pr` to be `map[string]pinnedResult`.

Reply:
> I do see how this might simplify the weighted 20/80 split though.
Actually, depending on the approach I think this weighting could be simplified.